File: 백준/Silver/2508. 사탕 박사 고창영/사탕 박사 고창영.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	static StringTokenizer st;
	private static int t;
	private static int r;
	private static int c;
	
	public static void main(String[] args) throws NumberFormatException, IOException {
		t = Integer.parseInt(br.readLine());
		
		for (int testcase = 0; testcase < t; testcase++) {
			br.readLine(); // 빈줄
			st= new StringTokenizer(br.readLine());
			
			r = Integer.parseInt(st.nextToken());
			c = Integer.parseInt(st.nextToken());
			
			char[][] lst = new char[r][c];
			
			for (int i = 0; i < r; i++) {
				String s = br.readLine();
				for (int j = 0; j < c; j++) {
					lst[i][j] = s.charAt(j);
				}
			}
			
			int answer = 0;
			for (int i = 0; i < r; i++) {
				for (int j = 0; j < c; j++) {
					if (lst[i][j] == 'o') {
						if (isValid(i, j - 1) && lst[i][j - 1] == '>') 
							if (isValid(i, j + 1) && lst[i][j + 1] == '<')
								answer++;
						if (isValid(i - 1, j) && lst[i - 1][j] == 'v') 
							if (isValid(i + 1, j) && lst[i + 1][j] == '^')
								answer++;
					}
				}
			}
			
			System.out.println(answer);
		}
	} // end of main

	private static boolean isValid(int y, int x) {
		return 0 <= y && y < r && 0 <= x && x < c;
	} // end of isValid
} // end of class

File: 백준/Silver/2508. 사탕 박사 고창영/사탕 박사 고창영.py
"""
완전탐색

o를 찾는다면 왼쪽에서 > 오른쪽에서 <
위쪽에서 ^ 아래쪽에서 v가 있는지 확인하기
"""

import sys

input = lambda: sys.stdin.readline().strip()

t = int(input())


def is_valid(y, x):
    return 0 <= y < r and 0 <= x < c


for _ in range(t):
    _ = input()
    r, c = map(int, input().split())
    lst = []
    answer = 0
    for i in range(r):
        lst.append(list(input()))

    for i in range(r):
        for j in range(c):
            if lst[i][j] == 'o':
                if is_valid(i, j - 1) and lst[i][j - 1] == '>':
                    if is_valid(i, j + 1) and lst[i][j + 1] == '<':
                        answer += 1
                if is_valid(i - 1, j) and lst[i - 1][j] == 'v':
                    if is_valid(i + 1, j) and lst[i + 1][j] == '^':
                        answer += 1

    print(answer)